### ISSUE TITLE
Screenshot urls do not work because dropbox has removed the public folder feature

### ISSUE THREAD

**Issue by gardiner** [fwenzel/upshot#77]
Dropbox has removed the public folder feature in favor of custom shared files and folders [1], so now the generated urls of Upshot are broken. Is there any way to circumvent this?

[1] https://www.dropbox.com/help/16

**Comment by fwenzel** [comment]
Yes I saw. This is breaking the core functionality of UpShot, which is very sad.

I'll edit the README file to make this clear.

I don't think there's a way to circumvent this in Dropbox-land. I am thinking UpShot should evolve into a tool that can either upload to *any number of different providers*, or at least to one that works well in the interim (imgur?).

What do you think?

**Comment by gardiner** [comment]
Thanks for your response!

Actually I think the simplicity is the best thing about upshot. Personally I think it'd be best if upshot could use something like the dropbox API [1] to create shareable links to the screenshots. This way everything would feel almost the same ;)

[1] https://www.dropbox.com/developers-v1/core/docs#shares

**Comment by fwenzel** [comment]
I like that a lot, though keep in mind that this functionality exists inside Dropbox already, so unless it is doing something different than Dropbox itself, it is not adding value (previously the different thing was a direct link to the image)

**Comment by gardiner** [comment]
Oh I probably didn't explain it right. I mean:

- user takes a screenshot using system shortcut
- upshot takes the screenshot file, optionally renames it, and moves it into dropbox directory
- upshot uses the dropbox api to create a public link for the file and copies it into clipboard

So yeah, each step in itself can be done without upshot, but the value is in integrating them seemlessly :) At least that's what I think.

**Comment by pavelhlushchanka** [comment]
@gardiner the value of upshot is that it returned a direct link to the image. What you explained above is already done by official dropbox app.

**Comment by gardiner** [comment]
@codenamestif Wow please show me how! Right now I always do a right click and tell dropbox to copy a shareable link into the clipboard. If this is in fact not necessary because dropbox can do this automatically and on its own I'd be thrilled :)

**Comment by pavelhlushchanka** [comment]
@gardiner [here you go](https://www.dropbox.com/en/help/1964)

**Comment by gardiner** [comment]
@codenamestif Wow. I'm ... baffled. I did not know that. Firstly: Thanks for the explanation, greatly appreciated. But secondly - what _exactly_ was/is the advantage of upshot over this functionality? It feels ... pretty much identical.

**Comment by pavelhlushchanka** [comment]
@gardiner upshot copies a direct link to the jpg/png image. Dropbox copies a link to the page with the image and that's not really convenient for sharing in different services.

**Comment by gardiner** [comment]
@codenamestif I see, thanks. Actually your link and your explanation solved this for me, so I'm closing the issue. Thanks again to both of you :) Keep up the great work!

**Comment by fwenzel** [comment]
Actually, I'll reopen this because the question (for me) is how I might fix this going forward. Using dropbox is not useful anymore, but I think (I hope you agree) that UpShot is a nice tool anyway, so if there's another way to get a direct link to the image, that'd be a good feature to replace it with.

**Comment by fwenzel** [comment]
Oh and by the way, you ask as if UpShot duplicated Dropbox functionality. It's actually the other way around. Dropbox added this functionality later.

**Comment by gardiner** [comment]
@fwenzel I totally agree! :) Thanks again.

**Comment by ghost** [comment]
Add support for S3 and all that sort of services? Don't know about others but maybe look into Google Drive, OneDrive all that stuff.

**Comment by fwenzel** [comment]
Thank you for the advice. I'll have to look into how I can implement this with a meaningful architecture. I might make it happen in the backend first (configured by something like a `~/.upshotrc` file) and do the frontend later. Open Source baby! ;)
